tests/ota/test_ota_matching.py

```diff
@@ -146,16 +146,14 @@ async def test_ota_matching_priority(tmp_path: pathlib.Path) -> None:
     ota.register_provider(SelfContainedProvider(index))
     ota.register_provider(BrokenProvider(index))
 
-    image1 = await ota.get_ota_image(device, query_cmd)
-    assert image1 is not None
-    assert image1 == zigpy.ota.OtaImageWithMetadata(
+    images1 = await ota.get_ota_images(device, query_cmd)
```

Comment:
Would it make sense to also assert the length of `images1.upgrades` == 1 here to make sure we don't get any other unwanted images later in the list?

E.g. (second line):
```suggestion
    images1 = await ota.get_ota_images(device, query_cmd)
    assert len(images1.upgrades) == 1
```

Reply:
With the way OTA is structured, four images in this test will match, just one of them has a higher specificity given that it has more metadata to match against.